scr/main/java/bittercode/util/DBUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.bittercode.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.model.StoreException;
+
+public class DBUtil {
+
+    private static Connection connection;
+
+    static {
+
+        try {
+
+            Class.forName(DatabaseConfig.DRIVER_NAME);
+            
+            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
+                    DatabaseConfig.DB_PASSWORD);
+        } catch (SQLException | ClassNotFoundException e) {
+
+            e.printStackTrace();

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Improve exception handling in static block</b></div>
<div id="fix">

The exception handling in the static block silently swallows exceptions by only printing the stack trace. This could mask critical database connectivity issues. Similar issues were also found in:
- scr/main/java/bittercode/service/impl/BookServiceImpl.java (line 83-85)
- scr/main/java/bittercode/service/impl/BookServiceImpl.java (line 58-60)
- scr/main/java/bittercode/service/impl/BookServiceImpl.java (line 167-169)
- scr/main/java/bittercode/service/impl/UserServiceImpl.java (line 48-49)
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,1 +1,2 @@
 +private static final Logger logger = LoggerFactory.getLogger(DBUtil.class);
 @@ -22,5 +22,6 @@
 -            e.printStackTrace();
 +            logger.error("Failed to initialize database connection", e);
 +            throw new RuntimeException("Database initialization failed", e);
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2514333017>#855223</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged